user/zig/template.py

```diff
@@ -24,14 +24,13 @@
 maintainer = "psykose <alice@ayaya.dev>"
 license = "MIT"
 url = "https://github.com/ziglang/zig"
-source = f"https://ziglang.org/download/{pkgver}/zig-{pkgver}.tar.xz"
-sha256 = "72014e700e50c0d3528cef3adf80b76b26ab27730133e8202716a187a799e951"
+source = f"https://ziglang.org/builds/zig-{pkgver}-dev.2631+3069669bc.tar.xz"
+sha256 = "395688a4a30e0afad2c6e2d0a4bded5d61c58d88d2e89362716732cb1c91b332"
 # lighten up the build, only applies to bootstrap
 hardening = ["!int", "!scp", "!var-init"]
 # lto only gets applied to the C bootstrap and slows down the build (doesn't
 # affect the zig output)
 options = ["!lto"]
-restricted = "work in progress"
```

Comment:
half of the WIP for zig itself (the other half being the build style you've now implemented) is that currently we don't really have plans to have multiple llvm versions at once. the current unwritten requirement for things linking libLLVM (zig also uses libclang and liblld but that's just fluff) is that they are generally ready by the time .3/.4 is out or so (when we upgrade to new majors). so far, zig isn't up to date with the latest llvm in a shipped release near these points (though i do see that on master it happens very fast). not sure how this is best resolved though. maybe eventually we'll just have to have multiple versions..

Reply:
There is another possibility that we can explore, which is to take advantage of zig's C backend, and support for [building from source without LLVM](https://github.com/ziglang/zig/?tab=readme-ov-file#building-from-source-without-llvm).

Such a build of zig is not suitable for direct users of zig, however, it could be used by the distro to package zig applications by compiling them to C code and then compiling and linking that C code with the system toolchain.

Reply:
hmm.. if that actually works for all regular zig projects one would want in a distro-packaged way (river, ncdu2, ..) then it honestly doesn't seem like the worst idea :D. i had no idea it was possible

Reply:
I haven't tried that approach yet but it certainly sounds promising in theory. I suspect there may be some rough edges that need to be ironed out but that's why I opened this PR in the first place :)